src/protocols/user_directed_commissioning/UserDirectedCommissioning.h

```diff
@@ -0,0 +1,192 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines objects for a CHIP Echo unsolicitied
+ *      initaitor (client) and responder (server).
+ *
+ */
+
+#pragma once
+
+#include "UDCClients.h"
+#include <core/CHIPCore.h>
+#include <mdns/Resolver.h>
+#include <messaging/ExchangeContext.h>
+#include <messaging/ExchangeMgr.h>
+#include <messaging/Flags.h>
+#include <protocols/Protocols.h>
+#include <support/CodeUtils.h>
+#include <support/DLLUtil.h>
+#include <support/logging/CHIPLogging.h>
+#include <transport/TransportMgr.h>
+
+namespace chip {
+namespace Protocols {
+namespace UserDirectedCommissioning {
+
+/**
+ * User Directed Commissioning Protocol Message Types
+ */
+enum class MsgType : uint8_t
+{
+    IdentificationDeclaration = 0x00,
+};
+
+class DLL_EXPORT InstanceNameResolver
+{
+public:
+    /**
+     * @brief
+     *   Called when a UDC message is received specifying the given instanceName
+     * This method indicates that UDC Server needs the Commissionable Node corresponding to
+     * the given instance name to be found. UDC Server will wait for OnCommissionableNodeFound.
+     *
+     * @param instanceName DNS-SD instance name for the client requesting commissioning
+     *
+     */
+    virtual void FindCommissionableNode(char * instanceName) {}
```

Comment:
Shouldn't this be pure virtual?  When would it be ok for this resolver to not implement this method at all?

Reply:
addressed in latest changes